node_modules/oae-authentication/lib/strategies/shibboleth/init.js

```diff
@@ -84,6 +84,11 @@ module.exports = function(config) {
                 'authoritative': true
             };
 
+            // Set users whose name resembles a Shibboleth identifier as private (eg. starts with http://)
+            if (displayName.indexOf('http://') === 0 || displayName.indexOf('shibboleth!') != -1) {
+                opts.visibility ='private';
+            } 
+
```

Comment:
When the user has an invalid display name / email address, the `editprofile` widget will be displayed in the UI. At that point, the user's account visibility should not show as private, but as the default profile visibility for that tenant, so once they save they get the default visibility (unless they change it).